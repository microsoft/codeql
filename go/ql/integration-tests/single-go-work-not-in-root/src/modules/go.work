go 1.23.0

toolchain go1.23.2

use (
	./subdir1
	./subdir2
)
