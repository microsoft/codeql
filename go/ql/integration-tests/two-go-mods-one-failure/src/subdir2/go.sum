github.com/microsoft/go-mssqldb v0.12.0 h1:nuQ0ygjq+dPZx78vkGH98aXZsk8tIdWHJaFV7ydhnqs=
