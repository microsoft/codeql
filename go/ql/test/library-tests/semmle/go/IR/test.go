package test

import (
	"fmt"
)

func testChannel() {
	var ch chan bool
	got, ok := <-ch
	fmt.Printf("%v %v", got, ok)
}

func testMap() {
	var m map[string]string
	got, ok := m["key"]
	fmt.Printf("%v %v", got, ok)
}

func testTypeAssert() {
	var i interface{}
	got, ok := i.(string)
	fmt.Printf("%v %v", got, ok)
}

func f(x, y int)    {}
func g() (int, int) { return 0, 0 }

func testNestedFunctionCalls() {
	f(g())

	// Edge case: when we call a function from a variable, `getTarget()` is not defined
	v := g
	f(v())
}
