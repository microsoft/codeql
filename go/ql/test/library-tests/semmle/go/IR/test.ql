import go

from IR::ExtractTupleElementInstruction extract, IR::Instruction base, int idx, Type resultType
where extract.extractsElement(base, idx) and resultType = extract.getResultType()
select extract, base, idx, resultType
