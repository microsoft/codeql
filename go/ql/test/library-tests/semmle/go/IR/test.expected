| test.go:9:2:9:16 | ... := ...[0] | test.go:9:13:9:16 | <-... | 0 | file://:0:0:0:0 | bool |
| test.go:9:2:9:16 | ... := ...[1] | test.go:9:13:9:16 | <-... | 1 | file://:0:0:0:0 | bool |
| test.go:15:2:15:20 | ... := ...[0] | test.go:15:13:15:20 | index expression | 0 | file://:0:0:0:0 | string |
| test.go:15:2:15:20 | ... := ...[1] | test.go:15:13:15:20 | index expression | 1 | file://:0:0:0:0 | bool |
| test.go:21:2:21:22 | ... := ...[0] | test.go:21:13:21:22 | type assertion | 0 | file://:0:0:0:0 | string |
| test.go:21:2:21:22 | ... := ...[1] | test.go:21:13:21:22 | type assertion | 1 | file://:0:0:0:0 | bool |
| test.go:29:2:29:7 | call to f[0] | test.go:29:4:29:6 | call to g | 0 | file://:0:0:0:0 | int |
| test.go:29:2:29:7 | call to f[1] | test.go:29:4:29:6 | call to g | 1 | file://:0:0:0:0 | int |
| test.go:33:2:33:7 | call to f[0] | test.go:33:4:33:6 | call to v | 0 | file://:0:0:0:0 | int |
| test.go:33:2:33:7 | call to f[1] | test.go:33:4:33:6 | call to v | 1 | file://:0:0:0:0 | int |
